client/components/theme/index.jsx

```diff
@@ -314,23 +321,28 @@ export class Theme extends Component {
 	};
 
 	renderBadge = () => {
-		const isLockedStyleVariation =
-			this.props.shouldLimitGlobalStyles &&
-			! isDefaultGlobalStylesVariationSlug( this.props.selectedStyleVariation?.slug );
+		const { selectedStyleVariation, shouldLimitGlobalStyles, siteId, siteSlug, theme, themeType } =
+			this.props;
+
+		const isPremiumTheme = isEnabled( 'themes/tiers' )
+			? theme.theme_tier?.slug === getPlan( PLAN_PREMIUM ).getPathSlug()
```

Comment:
You should [use the `PREMIUM_THEME` constant from `@automattic/design-picker`](https://github.com/Automattic/wp-calypso/blob/1f7dfda25ffe634c9132039bbbb62e2558f097c9/packages/design-picker/src/constants.ts#L41), since that's what we use to identify the slug of the Premium/Explorer tier.

`getPlan( PLAN_PREMIUM ).getPathSlug()` returns the slug of the plan which might or might not be the same as the tier slug (we're actually planning to rename the tier slug, see p1706110830735959-slack-C04DZ8M0GHW).

Reply:
Right, addressed on the last commit, sorry that you are having to review so many times 😅